src/compiler/scala/tools/nsc/Reporting.scala

```diff
@@ -544,16 +546,21 @@ object Reporting {
           }
         }
       } else if (s.startsWith("src=")) {
-        var pat = s.substring(4)
-        if (!pat.startsWith("/")) pat = "/" + pat
-        if (!pat.endsWith("$")) pat = pat + "$"
-        regex(pat).map(SourcePattern)
+        val arg = s.substring(4)
+        var pat = new mutable.StringBuilder()
+        if (rootDir.nonEmpty) pat += '^' ++= rootDir
```

Comment:
Would it not be easier to first relativize the path, then normalize filename separators to `/` and finally match against the pattern instead of modifying the regex itself?

Reply:
Either way seems fine with me, so I'll keep this as-is. We're building a new regex anyways.